pkg/stats/collector.go

```diff
@@ -179,6 +179,11 @@ func (s *BufferedCollector) Run(ctx context.Context) {
 			s.cache = make(keyIndex)
 			go s.send(metrics) // no need to block on this
```

Comment:
Think the issue is here to - sending using a background goroutine doesn't make sure the metrics are sent.
We can use the context of this goroutine to post the data and keep a large buffer for our channel.

Reply:
Handled that now.